app/controllers/api/projects.php

```diff
@@ -121,40 +126,56 @@
-                    'signed' => $attribute['signed'],
-                    'array' => $attribute['array'],
-                    'filters' => $attribute['filters'],
-                    'default' => $attribute['default'] ?? null,
-                    'format' => $attribute['format'] ?? ''
-                ]);
-            }
+            go(function () use ($collection, $register, $key, $project) {
+                $db = $register->get('dbPool')->get();
+                $redis = $register->get('redisPool')->get();
+                $cache = new Cache(new Redis($redis));
 
-            foreach ($collection['indexes'] as $index) {
-                $indexes[] = new Document([
-                    '$id' => $index['$id'],
-                    'type' => $index['type'],
-                    'attributes' => $index['attributes'],
-                    'lengths' => $index['lengths'],
-                    'orders' => $index['orders'],
-                ]);
-            }
+                $projectDB = new Database(new MariaDB($db), $cache);
+                $projectDB->setDefaultDatabase(App::getEnv('_APP_DB_SCHEMA', 'appwrite'));
+                $projectDB->setNamespace("_{$project->getInternalId()}");
+
+                if (($collection['$collection'] ?? '') !== Database::METADATA) {
+                    return;
+                }
+                $attributes = [];
+                $indexes = [];
+
+                foreach ($collection['attributes'] as $attribute) {
+                    $attributes[] = new Document([
+                        '$id' => $attribute['$id'],
+                        'type' => $attribute['type'],
+                        'size' => $attribute['size'],
+                        'required' => $attribute['required'],
+                        'signed' => $attribute['signed'],
+                        'array' => $attribute['array'],
+                        'filters' => $attribute['filters'],
+                        'default' => $attribute['default'] ?? null,
+                        'format' => $attribute['format'] ?? ''
+                    ]);
+                }
+
+                foreach ($collection['indexes'] as $index) {
+                    $indexes[] = new Document([
+                        '$id' => $index['$id'],
+                        'type' => $index['type'],
+                        'attributes' => $index['attributes'],
+                        'lengths' => $index['lengths'],
+                        'orders' => $index['orders'],
+                    ]);
+                }
 
-            $dbForProject->createCollection($key, $attributes, $indexes);
+                $projectDB->createCollection($key, $attributes, $indexes);
+                $register->get('dbPool')->put($db);
+                $register->get('redisPool')->put($redis);
+            });
         }
 
+        Console::info('Project Successfully created, Generated in ' . (\microtime(true) - $time) * 1000 . 'ms');
+
```

Comment:
We might have to wait here until the co-routines for the collections are finished here. Otherwise we return a response while it's still creating when the db is slow.

Reply:
Is there a way we can do something like promise.all without co\run? Since co\run won't work due to swoole attempting to create a new scheduler. Otherwise we may have to use a sleep

Reply:
You need to use `WaitGroup` 👍🏻

Reply:
Implemented 👍